src/endpoints/common_endpoint_registry.cpp

```diff
@@ -68,29 +70,91 @@ namespace ccf
-      {
-        GetCommit::Out out;
-        out.transaction_id.view = view;
-        out.transaction_id.seqno = seqno;
-        return make_success(out);
-      }
-      else
+      auto result = get_last_committed_txid_v1(view, seqno);
+      if (result != ccf::ApiResult::OK)
       {
         return make_error(
           HTTP_STATUS_INTERNAL_SERVER_ERROR,
           ccf::errors::InternalError,
           fmt::format("Error code: {}", ccf::api_result_to_str(result)));
       }
+
+      GetCommit::Out out;
+      out.transaction_id.view = view;
+      out.transaction_id.seqno = seqno;
+
+      // Parse arguments from query
+      const auto parsed_query =
+        http::parse_query(ctx.rpc_ctx->get_request_query());
+
+      std::string error_reason;
+      std::string view_history;
+      http::get_query_value(
+        parsed_query, view_history_param_key, view_history, error_reason);
+      // continue even if there is an error as the view_history param is
+      // optional
+
+      if (view_history == "true")
+      {
+        std::vector<ccf::TxID> history;
+        result = get_view_history_v1(history);
+        if (result != ccf::ApiResult::OK)
+        {
+          return make_error(
+            HTTP_STATUS_INTERNAL_SERVER_ERROR,
+            ccf::errors::InternalError,
+            fmt::format("Error code: {}", ccf::api_result_to_str(result)));
+        }
+        out.view_history = history;
+      }
+      else
+      {
+        std::string error_reason;
+        ccf::View view_history_since;
+        http::get_query_value(
+          parsed_query,
+          view_history_since_param_key,
+          view_history_since,
+          error_reason);
+        // continue even if there is an error as the view_history param is
+        // optional
+
+        if (error_reason == "")
+        {
+          std::vector<ccf::TxID> history;
+          result = get_view_history_v1(history, view_history_since);
+          if (result == ccf::ApiResult::InvalidArgs)
+          {
+            return make_error(
+              HTTP_STATUS_BAD_REQUEST,
+              ccf::errors::InvalidQueryParameterValue,
+              fmt::format("Error code: {}", ccf::api_result_to_str(result)));
+          }
+          else if (result != ccf::ApiResult::OK)
+          {
+            return make_error(
+              HTTP_STATUS_INTERNAL_SERVER_ERROR,
+              ccf::errors::InternalError,
+              fmt::format("Error code: {}", ccf::api_result_to_str(result)));
+          }
+          out.view_history = history;
+        }
+        else
+        {
+          CCF_APP_FAIL("Failed to get view history since: {}", error_reason);
+        }
+      }
+
+      return make_success(out);
     };
     make_command_endpoint(
       "/commit", HTTP_GET, json_command_adapter(get_commit), no_auth_required)
       .set_auto_schema<GetCommit>()
+      .add_query_parameter<bool>(view_history_param_key)
+      .add_query_parameter<ccf::View>(view_history_since_param_key)
```

Comment:
I think these should be optional not to break compatibility with previous versions?

Reply:
Sneaky default params.